subsys/net/lib/download_client/src/download_client.c

```diff
@@ -103,6 +103,31 @@ static int socket_sectag_set(int fd, int sec_tag)
 	return 0;
 }
 
+static int socket_tls_hostname_set(int fd, const char * const hostname)
+{
+	if (!hostname) {
+		return -EINVAL;
+	}
+
+	char parsed_host[HOSTNAME_SIZE];
+	int err = url_parse_host(hostname, parsed_host, sizeof(parsed_host));
```

Comment:
nit: please split declaration and assignment to be consistent with the rest of the file